[14.0][FIX] account_financial_report: Trial Balance with 'hide account at 0'

ernestotejeda opened the pull request:
Cc @Tecnativa TT34490 TT35127

Fix 'Trial Balance' report: The 'Trial Balance' report shows accounts at 0 even if 'Hide accounts at 0' option is checked.

Commit:
[FIX] account_financial_report: Trial Balance with 'hide account at 0'

ernestotejeda commented:
ping @pedrobaeza @sergio-teruel @carlosdauden  @JoanSForgeFlow

ernestotejeda commented:
ping @pedrobaeza

pedrobaeza commented on account_financial_report/report/trial_balance.py:
```diff
@@ -300,6 +300,29 @@ def _compute_partner_amount(
+    def _remove_accounts_at_cero(self, total_amount, show_partner_details):
+        def is_removable(d):
+            return (
+                d["initial_balance"] == 0
```
This should be used with `float_is_zero` (and if not possible to get decimal precision, put 8 digits).

pedrobaeza commented on account_financial_report/report/trial_balance.py:
```diff
@@ -300,6 +300,29 @@ def _compute_partner_amount(
+            elif show_partner_details:
+                partner_to_remove = []
+                for key, value in ta_data.items():
+                    if isinstance(key, int) and is_removable(value):
```
Why do you need to do this `isinstance` thing? Always explain as comment the weird things.

ernestotejeda replied:
I have put a comment in the code, although it is difficult for me to explain. The image below shows the value of the total_amount variable (the main dict used for printing the report) while debugging the code in a use case. If the show_partner_details option is checked, in addition to the "initial_balance" ... "ending_balance" keys (marked in red), the same account data dictionary will also have the partner IDs as key (marked in blue) with their respective data. That's why i used isinstance method. Check the comment i have put.

![image](https://user-images.githubusercontent.com/38267832/156811846-8dc40f82-dd1f-4fa4-bd36-18bfebde6370.png)

And this is the result with this data and without my code:

![image](https://user-images.githubusercontent.com/38267832/156814049-b42d87f7-5f95-4403-be6f-5f45925069b1.png)

Shide commented:
@pedrobaeza are you ok with the new changes?

rafaelbn reviewed (approved):
Tested functionally 👍 

https://github.com/OCA/account-financial-reporting/blob/14.0/account_financial_report/wizard/trial_balance_wizard.py#L55-L57

```
    hide_account_at_0 = fields.Boolean(
        string="Hide accounts at 0",
        default=True,
        help="When this option is enabled, the trial balance will "
        "not display accounts that have initial balance = "
        "debit = credit = end balance = 0",
    )
```

Shide commented:
@pedrobaeza are you ok with the changes (comments) of @ernestotejeda ?

pedrobaeza reviewed (changes requested):
It's not working properly yet.

Commit:
[FIX] account_financial_report: trial balance hierarchy

When the report is generated with 'hierarchy' option checked,
the group hierarchy values are not calculated correctly

ernestotejeda commented:
@pedrobaeza could please you check again with this new commit?

pedrobaeza commented:
@ernestotejeda can you remove the "computed hierarchy" option as now the hierarchy is already computed in the records and no need to compute them on the fly?

Commit:
[IMP] account_financial_report: Remove 'Computed Accounts' option from 'trial balance'.

ernestotejeda commented:
> @ernestotejeda can you remove the "computed hierarchy" option as now the hierarchy is already computed in the records and no need to compute them on the fly?

@pedrobaeza change done

ernestotejeda commented:
Travis is red, i will check it tomorrow

ernestotejeda commented:
@pedrobaeza travis is green now

pedrobaeza commented:
/ocabot merge minor

Please fw-port it to v15
